src/hardware/mixer.cpp

```diff
@@ -419,27 +419,51 @@ void MixerChannel::AddSilence()
 	MIXER_UnlockAudioDevice();
 }
 
+static const std::map<FilterState, std::string> filter_state_map = {
+        {FilterState::Off, "disabled"},
+        {FilterState::On, "enabled"},
+        {FilterState::ForcedOn, "enabled (forced)"},
+};
+
+void MixerChannel::SetHighPassFilter(const FilterState state)
+{
+	filters.highpass.state = state;
+
+	auto it = filter_state_map.find(state);
+	if (it != filter_state_map.end()) {
+		auto filter_state = it->second;
+		LOG_MSG("MIXER: %s channel highpass filter %s",
+		        name.c_str(),
+		        filter_state.c_str());
+	}
+}
+
 void MixerChannel::SetLowPassFilter(const FilterState state)
 {
-	filter.state = state;
+	filters.lowpass.state = state;
 
-	static const std::map<FilterState, std::string> filter_state_map = {
-	        {FilterState::Off, "disabled"},
-	        {FilterState::On, "enabled"},
-	        {FilterState::ForcedOn, "enabled (forced)"},
-	};
 	auto it = filter_state_map.find(state);
 	if (it != filter_state_map.end()) {
 		auto filter_state = it->second;
-		LOG_MSG("MIXER: %s channel filter %s", name.c_str(), filter_state.c_str());
+		LOG_MSG("MIXER: %s channel lowpass filter %s",
+		        name.c_str(),
+		        filter_state.c_str());
 	}
 }
 
+void MixerChannel::ConfigureHighPassFilter(const uint8_t order,
+                                           const uint16_t cutoff_freq)
+{
+	assert(order > 0 && order <= max_filter_order);
+	for (auto &f : filters.highpass.hpf)
+		f.setup(order, mixer.sample_rate, cutoff_freq);
+}
```

Comment:
Really nice to read.
The asserts help me understand what you expect to be the normal range. 
Hard verification doubling up documentation! Good stuff.

Reply:
Yeah, I prefer the code-as-living-documentation approach these days.